compiler/noirc_frontend/src/hir_def/types.rs

```diff
@@ -956,6 +1003,15 @@ impl Type {
         matches!(self.follow_bindings(), Type::Integer(_, _))
     }
 
+    fn is_field_element(&self) -> bool {
+        match self.follow_bindings() {
+            Type::FieldElement => true,
+            Type::TypeVariable(var) => var.is_field_element(),
+            Type::Constant(_, kind) => kind.is_field_element(),
```

Comment:
It seems odd to return true for both Type::FieldElement and type-level constants of type Field. Do we need both?

Reply:
Both are currently used, since checking whether the `Type::Constant`'s `Kind` is a field element uses the check for `Type::FieldElement`. I'll split the two versions apart to be more clear.